components/menu/test/menu.vdiff.js

```diff
@@ -0,0 +1,242 @@
+				`);
+				if (action) await action(elem);
+				await expect(elem).to.be.golden();
+			});
+		});
+
+		it('rtl', async() => {
+			const elem = await fixture(html`
+				<d2l-menu>
+					<d2l-menu-item text="a"></d2l-menu-item>
+					<d2l-menu-item text="b"></d2l-menu-item>
+				</d2l-menu>
+			`, { rtl: true });
+			await expect(elem).to.be.golden();
+		});
+	});
+
+	describe('disabled', () => {
+		[
+			{ name: 'simple' },
+			{ name: 'hover', action: elem => hoverElem(elem.querySelector('d2l-menu-item')) },
+			{ name: 'focus', action: elem => focusElem(elem.querySelector('d2l-menu-item')) }
+		].forEach(({ name, action }) => {
+			it(name, async() => {
+				const elem = await fixture(html`
+					<d2l-menu>
+						<d2l-menu-item text="a" disabled></d2l-menu-item>
+					</d2l-menu>
+				`);
+				if (action) await action(elem);
+				await expect(elem).to.be.golden();
+			});
+		});
+	});
+
+	describe('dark', () => {
+		[
+			{ name: 'simple' },
+			{ name: 'hover', action: elem => hoverElem(elem.querySelector('d2l-menu-item[text="normal"]')) },
+			{ name: 'focus', action: elem => focusElem(elem.querySelector('d2l-menu-item[text="normal"]')) }
+		].forEach(({ name, action }) => {
+			it(name, async() => {
+				const elem = await fixture(html`
+					<d2l-menu theme="dark">
+						<d2l-menu-item text="nested">
+							<d2l-menu theme="dark">
+								<d2l-menu-item text="a3"></d2l-menu-item>
+							</d2l-menu>
+						</d2l-menu-item>
+						<d2l-menu-item text="normal"></d2l-menu-item>
+						<d2l-menu-item text="disabled" disabled></d2l-menu-item>
+						<d2l-menu-item-separator></d2l-menu-item-separator>
+						<d2l-menu-item-link text="link" href="https://en.wikipedia.org/wiki/Universe"></d2l-menu-item-link>
+						<d2l-menu-item text="supporting">
+							<div slot="supporting">supporting</div>
+						</d2l-menu-item>
+					</d2l-menu>
+				`);
+				if (action) await action(elem);
+				await expect(elem).to.be.golden();
+			});
+		});
+	});
+
+	describe('nested', () => {
+		const nestedTemplate = html`
+			<d2l-menu>
+				<d2l-menu-item text="b">
+					<d2l-menu id="nested-menu">
+						<d2l-menu-item text="b2"></d2l-menu-item>
+					</d2l-menu>
+				</d2l-menu-item>
+			</d2l-menu>
+		`;
+		const nestedLongTemplate = html`
+			<d2l-menu>
+				<d2l-menu-item id="first-item" text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.">
+					<d2l-menu>
+						<d2l-menu-item text="a3"></d2l-menu-item>
+					</d2l-menu>
+				</d2l-menu-item>
+				<d2l-menu-item text="b"></d2l-menu-item>
+				<d2l-menu-item text="c"></d2l-menu-item>
+			</d2l-menu>
+		`;
+		const customViewTemplate = html`
+			<d2l-menu>
+				<d2l-menu-item text="a">
+					<d2l-custom-view></d2l-custom-view>
+				</d2l-menu-item>
+			</d2l-menu>
+		`;
+
+		[true, false].forEach(rtl => {
```

Comment:
Unfortunately had to hardcode sizes in a number of the tests below, just like for `filter`.  `hierarchical-view` is extremely hard to detect when it's done and the menu is done resizing for the last time.